### PR TITLE
UW 203 User interface for handling config files

### PR DESCRIPTION
**Pull request by j-derrico**
**Description**
[Fixes Issue #151 ](https://github.com/ufs-community/workflow-tools/issues/151)

The config parsing tool (probably set_config.py) needs a command line interface for user interaction. It should provide all the standard options as outlined in the [Toolbox MVP](https://confluence-epic.woc.noaa.gov/display/UW/Toolbox+MVP) and should aim at achieving the following behaviors:

-i to accept an input "base" config file (required)
-o to write a rendered output file (optional)
-c to accept a user config file (optional)
These files will need some way of knowing what type of file we give it. As a first pass, I think that determining type from the suffix is sufficient. We'll likely want a better solution at a future date.

Assume the output file type is the same as the input file type in this ticket.

The end purpose of the tool is to update key/value pairs, concatenate sections or key/value pairs, compare two fully formed config files, and transform an input base value to a different output configure type.  We have tickets to handle the full range of these functions. Just do the input/output handling here.

**Type of change**

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [x] pytests in GitHub actions.
- [ ] Tests on XYZ OS/HPC/CSP
-->

**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation.  I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

File: src/uwtools/set_config.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        config_obj = config.YAMLConfig(user_args.input_base_file)
+
+    elif file_type in [".bash", ".sh", ".ini", ".IN"]:
+        config_obj = config.INIConfig(user_args.input_base_file)
+
+    elif file_type == ".nml":
+        config_obj = config.F90Config(user_args.input_base_file)
+
+    else:
+        print ("Set config failure: bad file type")
```
Comment (christinaholtNOAA) [inline src/uwtools/set_config.py:79]: Would you mind taking out this extra space after `print`? Totally just a weird syntax pet peeve of mine.

**Review by christinaholtNOAA — APPROVED** [review]
Nicely done!

It looks like the linter started failing, but not on the code committed here. We should prob open a GH Issue and take a look at that oddity, but let's move forward with this PR.

File: src/uwtools/set_config.py (hunk 1 above)
Comment (j-derrico) [reply src/uwtools/set_config.py]: It's been removed!

**Review by christinaholtNOAA — APPROVED** [review]

**Comment by WeirAE** [comment]
Adding comments from our quick discussion on Monday that were incorporated into the version that went into this PR: tests and linter ran fine, setup looked good. I noted some extra spaces in front of () that may need to be removed, but looked otherwise good. I requested that a fail condition be added if no appropriate suffix was found with a test, which were included here.

**Review by christinaholtNOAA — APPROVED** [review]

**Pull request merged**